### ISSUE TITLE
Yara doesn't print out anything when we try to scan `/proc/{pid}/maps`

### ISSUE THREAD

**Issue by leomao10** [VirusTotal/yara#1838]
**Describe the bug**
We try to use yara to scan `/proc/{pid}/maps` and found that nothing get print out even the rules maps strings in the files.

The rules we use is like this:
```
rule node_package {
    meta:
        name = "babelcli"

    strings:
        $name = "babelcli",
        $text = "node_modules"
    condition:
        all of them
}
```

yara command is like `yara ./rule.yara /proc/{id}/maps`, and nothing get print out. However, if I copy `maps` file to other location like `cp /proc/{id}/maps /{id}_maps`, and use similar command to scan the new file: `yara ./rule.yara /{id}_maps`, then there is output from yara with rule matched.

**To Reproduce**
- `docker container run -it --entrypoint /bin/bash ubuntu`
- run command `apt-get update; apt-get install yara procps` in container
- Create sleep_rule.yara in container and add following rule:
```
rule LOG_sleep {
    meta:
        name = "sleep process map"

    strings:
        $text1 = "ld-linux-aarch64.so.1" nocase
        $text2 = "aarch64-linux-gnu" nocase
    condition:
        all of them
}
```
- run `sleep 1000 &` in container and get its pid
- run `yara /sleep_rule.yara /proc/{id}/maps` in container and there is nothing prints out
- `cp /proc/{id}/maps /{id}_maps` and then run `yara /sleep_rule.yara /{id}_maps` and we see `LOG_sleep /502_maps` message print out.

**Expected behavior**
When using `yara /sleep_rule.yara /proc/{id}/maps`, it should print out the rule if yara rule match.

**Please complete the following information:**
 - OS: ubuntu
 - YARA version: 3.9.0, 4.2.1

**Comment by hillu** [comment]
YARA wants to memory-map the files it scans.
I was going to write that this is not possible at all with files in `/proc`, but the root cause for the silent failure is that YARA uses the file size as reported by _fstat(3)_ to determine the size it uses for the memory map – and the file size for any file in `/proc/<pid>/` is 0.

In my opinion,YARA should at least recognize files where using the usual fstat+mmap approach does not make sense and refuse to scan them.

**Comment by plusvic** [comment]
Closed in #1848
